src/main/java/io/github/thebusybiscuit/slimefun4/core/multiblocks/MultiBlockMachine.java

```diff
@@ -104,12 +104,20 @@ public void postRegister() {
     @Override
     public void load() {
         super.load();
-
-        for (ItemStack recipeItem : displayRecipes) {
-            SlimefunItem item = SlimefunItem.getByItem(recipeItem);
-
-            if (item == null || !item.isDisabled()) {
-                recipes.add(new ItemStack[] { recipeItem });
+        
+        for (ItemStack inputStack : displayRecipes) {
+            int index = displayRecipes.indexOf(inputStack);
```

Comment:
No need for this expensive `.indexOf` operation here, we are looping through this list anyway. So just create your own counter and increment it on every run.